velox/connectors/hive/storage_adapters/hdfs/tests/HdfsFileSystemTest.cpp

```diff
@@ -0,0 +1,331 @@
+    FAIL() << "expected VeloxException";
+  } catch (facebook::velox::VeloxException const& error) {
+    EXPECT_THAT(
+        error.message(),
+        testing::HasSubstr(
+            "hdfsPort is empty, configuration missing for hdfs port"));
+  }
+}
+
+TEST_F(HdfsFileSystemTest, missingFileViaReadFile) {
+  try {
+    struct hdfsBuilder* builder = hdfsNewBuilder();
+    hdfsBuilderSetNameNode(builder, "localhost");
+    hdfsBuilderSetNameNodePort(builder, 7878);
+    auto hdfs = hdfsBuilderConnect(builder);
+    HdfsReadFile readFile(hdfs, "/path/that/does/not/exist");
+    FAIL() << "expected VeloxException";
+  } catch (facebook::velox::VeloxException const& error) {
+    EXPECT_THAT(
+        error.message(),
+        testing::HasSubstr(
+            "Unable to get file path info for file: /path/that/does/not/exist. got error: FileNotFoundException: Path /path/that/does/not/exist does not exist."));
+  }
+}
+
+TEST_F(HdfsFileSystemTest, schemeMatching) {
+  try {
+    auto fs =
+        std::dynamic_pointer_cast<facebook::velox::filesystems::HdfsFileSystem>(
+            filesystems::getFileSystem("/", nullptr));
+    FAIL() << "expected VeloxException";
+  } catch (facebook::velox::VeloxException const& error) {
+    EXPECT_THAT(
+        error.message(),
+        testing::HasSubstr(
+            "No registered file system matched with filename '/'"));
+  }
+  auto fs =
+      std::dynamic_pointer_cast<facebook::velox::filesystems::HdfsFileSystem>(
+          filesystems::getFileSystem(fullDestinationPath, nullptr));
+  ASSERT_TRUE(fs->isHdfsFile(fullDestinationPath));
+}
+
+TEST_F(HdfsFileSystemTest, writeNotSupported) {
+  try {
+    facebook::velox::filesystems::registerHdfsFileSystem();
+    std::unordered_map<std::string, std::string> configurationValues(
+        {{"hdfs_host", "localhost"}, {"hdfs_port", "7878"}});
+    auto memConfig =
+        std::make_shared<const core::MemConfig>(std::move(configurationValues));
+    auto hdfsFileSystem =
+        filesystems::getFileSystem(fullDestinationPath, memConfig);
+    hdfsFileSystem->openFileForWrite("/path");
+  } catch (facebook::velox::VeloxException const& error) {
+    EXPECT_EQ(error.message(), "Write to HDFS is unsupported");
+  }
+}
+
+TEST_F(HdfsFileSystemTest, removeNotSupported) {
+  try {
+    facebook::velox::filesystems::registerHdfsFileSystem();
+    std::unordered_map<std::string, std::string> configurationValues(
+        {{"hdfs_host", "localhost"}, {"hdfs_port", "7878"}});
+    auto memConfig =
+        std::make_shared<const core::MemConfig>(std::move(configurationValues));
+    auto hdfsFileSystem =
+        filesystems::getFileSystem(fullDestinationPath, memConfig);
+    hdfsFileSystem->remove("/path");
+  } catch (facebook::velox::VeloxException const& error) {
+    EXPECT_EQ(error.message(), "Does not support removing files from hdfs");
+  }
+}
+
+TEST_F(HdfsFileSystemTest, multipleThreadsWithReadFile) {
```

Comment:
Isn't  `multipleThreadsWithReadFile` redundant given `multipleThreadsWithFileSystem`

Reply:
Got it. Thanks.